### PR TITLE
CI: Rename testing job

### LINKED ISSUE #236: Yapf prevent CI checks
We should try to change the CI such that wrong formatting does not skip the other steps of the CI, such as testing.

### PR DESCRIPTION
**Pull request by webmiche**
closes #236 

This PR renames the testing CI jobs to be more intuitive to prevent confusion.

### CODE REVIEW

File: .github/workflows/ci-core.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # This workflow will install Python dependencies, run tests and lint with a single version of Python
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: CI - Python application
+name: CI - Python-based Testing
```
Comment (georgebisbas) [inline .github/workflows/ci-core.yml:4]: CI - Python testing?

Comment (webmiche) [reply .github/workflows/ci-core.yml:4]: I am thinking of a future where we have a test that actually compiles MLIR, which I would then call C++-based testing. I think Python testing sounds like we are testing Python, whereas the thing I want to say is "test everything that purely runs in Python". What do you think?

Comment (georgebisbas) [reply .github/workflows/ci-core.yml:4]: I would love that, where so you see you that in the timeline? Roughly?

**Review by georgebisbas — APPROVED** [review]

File: .github/workflows/ci-core.yml (hunk 1 above)
Comment (webmiche) [reply .github/workflows/ci-core.yml:4]: I think I can look into getting the CI job running later this week.

My question was more about the name of this CI job (`Python Testing` vs `Python-based Testing`).

Comment (georgebisbas) [reply .github/workflows/ci-core.yml:4]: > I think I can look into getting the CI job running later this week.
That would be awesome!!!
 
> My question was more about the name of this CI job (`Python Testing` vs `Python-based Testing`).
Not a problem for me

**Pull request merged**